lib/web_ui/lib/src/engine/canvaskit/image.dart

```diff
@@ -376,7 +376,8 @@ class CkImage implements ui.Image, StackTraceDebugger {
     ui.ImageByteFormat format = ui.ImageByteFormat.rawRgba,
   }) {
     assert(_debugCheckIsNotDisposed());
-    if (videoFrame != null) {
+    // Web codecs do not support I420, I444, I422 videoFrame formats.
+    if (videoFrame != null && videoFrame!.format != 'I420' && videoFrame!.format != 'I444' && videoFrame!.format != 'I422') {
```

Comment:
Are we still leaving `readPixelsFromVideoFrame` because of the shared memory bug in `_readPixelsFromSkImage`?

Reply:
That is one reason. Another is that using wasm codecs `toByteData` with png requested format fails when CkBrowserImageDecoder is used. There is potentially an issue when the videoFrame format is I420, I444, I422 and a `toByteData` call is made with requested format png.